File: ERP-INVESTIGACION/articulos-frontend/src/app/features/almacen/service/unidad-medida.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { UnidadMedidaRequest, UnidadMedidaResponse } from '../../configuraciones/models/unidad-medida.model';
import { environment } from '../../../environments/environment';
import { ApiResponse } from '../../../shared/modals/response.model';

@Injectable({
    providedIn: 'root'
})
export class UnidadMedidaService {
    private apiUrl = `${environment.apiUrl}/unidades-medida`;

    constructor(private http: HttpClient) { }

    listarActivas(): Observable<ApiResponse<any>> {
        return this.http.get<ApiResponse<any>>(`${this.apiUrl}/activas?page=0&size=1000`);
    }

    listar(page: number, size: number): Observable<ApiResponse<any>> {
        return this.http.get<ApiResponse<any>>(`${this.apiUrl}?page=${page}&size=${size}`);
    }

    crear(request: UnidadMedidaRequest): Observable<ApiResponse<UnidadMedidaResponse>> {
        return this.http.post<ApiResponse<UnidadMedidaResponse>>(this.apiUrl, request);
    }

    actualizar(id: number, request: UnidadMedidaRequest): Observable<ApiResponse<UnidadMedidaResponse>> {
        return this.http.put<ApiResponse<UnidadMedidaResponse>>(`${this.apiUrl}/${id}`, request);
    }

    eliminar(id: number): Observable<ApiResponse<void>> {
        return this.http.delete<ApiResponse<void>>(`${this.apiUrl}/${id}`);
    }
}

File: ERP-INVESTIGACION/articulos-backend/src/main/java/com/pe/articulos/modules/catalogo/service/UnidadMedidaService.java
package com.pe.articulos.modules.catalogo.service;

import org.springframework.data.domain.Pageable;

import com.pe.articulos.core.shared.dto.PageResponse;
import com.pe.articulos.modules.catalogo.dto.UnidadMedidaRequest;
import com.pe.articulos.modules.catalogo.dto.UnidadMedidaResponse;

public interface UnidadMedidaService {

    PageResponse<UnidadMedidaResponse> listarActivas(Pageable pageable);

    PageResponse<UnidadMedidaResponse> listarTodos(Pageable pageable);

    UnidadMedidaResponse crear(UnidadMedidaRequest request);

    UnidadMedidaResponse actualizar(Long id, UnidadMedidaRequest request);

    void eliminar(Long id);
}
